### ISSUE TITLE
DuplicateAlias error

### ISSUE THREAD

**Issue by mcm-ham** [DockYard/postgres_ext-serializers#18]
If I serialize a class like:

```ruby
class Offer < ActiveRecord::Base
  belongs_to :created_by, class_name: 'User', inverse_of: :offers
  belongs_to :reviewed_by, class_name: 'User', inverse_of: :reviewed_offers
end
```

Or

```ruby
class Offer < ActiveRecord::Base
  belongs_to :user
  has_many :items
end

class Item < ActiveRecord::Base
  belongs_to :user 
end
```

Then it produces two CTE called `users_attributes_filter` which throws a PG DuplicateAlias error.

**Comment by mcm-ham** [comment]
Note #23 didn't completely fix it since you can have multiple `relationname_foreignkeyname_attributes_filter` with different containing tables so have changed approach to keep a cache of names and simply add an incrementing number if name already exists https://github.com/crossroads/postgres_ext-serializers/commit/89ccd2b8f492ee869f71fe7a942249e7b7ac242e

**Comment by felixbuenemann** [comment]
Merged in PR #38.
